<a href="https://colab.research.google.com/github/raquelc92/GAPE/blob/main/An%C3%A1lise_Dados_GAPE.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [ ]:
df = pd.read_csv('gape.csv')

In [ ]:
df.head()

NO_UF SG_UF  CO_UF NO_MUNICIPIO  CO_MUNICIPIO  CO_ENTIDADE  TP_DEPENDENCIA  \
0  Pará    PA     15   Pau D'Arco       1505551     15130959               3   
1  Pará    PA     15   Pau D'Arco       1505551     15131572               2   
2  Pará    PA     15   Pau D'Arco       1505551     15160467               3   
3  Pará    PA     15   Pau D'Arco       1505551     15528294               3   
4  Pará    PA     15   Pau D'Arco       1505551     15534847               3   

        REDE  TP_LOCALIZACAO  TP_SITUACAO_FUNCIONAMENTO  ...  Vl Devido Total  \
0  Municipal               2                          1  ...         2.451,00   
1   Estadual               1                          1  ...         3.328,00   
2  Municipal               2                          1  ...                -   
3  Municipal               2                          1  ...                -   
4  Municipal               2                          1  ...         2.451,00   

   Vl Ajuste Custeio  Vl Ajuste Capital  Vl Ajuste Total  \
0                  0                  0                0   
1                  0                  0                0   
2                  -                  -                -   
3                  -                  -                -   
4                  0                  0                0   

   Vl Final Devido Total  Vl Pago Custeio  Vl Pago Capital  Valor Pago Total  \
0               2.451,00         1.560,01           890,99          2.451,00   
1               3.328,00                0                0                 0   
2                      -                -                -                 -   
3                      -                -                -                 -   
4               2.451,00         1.560,01           890,99          2.451,00   

   Data Ord. Pgto  in_MED  
0      13/10/2021     NaN  
1             NaN      EM  
2               -     NaN  
3               -     NaN  
4      13/10/2021     NaN  

[5 rows x 82 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181 entries, 0 to 180
Data columns (total 82 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   NO_UF                      181 non-null    object 
 1   SG_UF                      181 non-null    object 
 2   CO_UF                      181 non-null    int64  
 3   NO_MUNICIPIO               181 non-null    object 
 4   CO_MUNICIPIO               181 non-null    int64  
 5   CO_ENTIDADE                181 non-null    int64  
 6   TP_DEPENDENCIA             181 non-null    int64  
 7   REDE                       181 non-null    object 
 8   TP_LOCALIZACAO             181 non-null    int64  
 9   TP_SITUACAO_FUNCIONAMENTO  181 non-null    int64  
 10  IN_DESKTOP_ALUNO           181 non-null    int64  
 11  QT_DESKTOP_ALUNO           181 non-null    int64  
 12  IN_COMP_PORTATIL_ALUNO     181 non-null    int64  
 13  QT_COMP_PORTATIL_ALUNO     181 non-null    int64  

In [ ]:
df.drop(["QT_MAT_BAS", "QT_MAT_BAS_D", "QT_MAT_BAS_N", "QT_MAT_INF", "QT_MAT_INF_CRE", "QT_MAT_INF_PRE",
         "QT_MAT_FUND", "QT_MAT_FUND_AF", "QT_MAT_MED"], axis=1, inplace=True)

In [ ]:
df.drop(["QT_MAT_PROF", "QT_MAT_PROF_TEC", "QT_MAT_EJA", "QT_MAT_EJA_FUND", "QT_MAT_EJA_MED", "QT_MAT_ESP", "QT_MAT_ESP_CC", "QT_MAT_ESP_CE"], axis=1, inplace=True)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181 entries, 0 to 180
Data columns (total 65 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   NO_UF                      181 non-null    object 
 1   SG_UF                      181 non-null    object 
 2   CO_UF                      181 non-null    int64  
 3   NO_MUNICIPIO               181 non-null    object 
 4   CO_MUNICIPIO               181 non-null    int64  
 5   CO_ENTIDADE                181 non-null    int64  
 6   TP_DEPENDENCIA             181 non-null    int64  
 7   REDE                       181 non-null    object 
 8   TP_LOCALIZACAO             181 non-null    int64  
 9   TP_SITUACAO_FUNCIONAMENTO  181 non-null    int64  
 10  IN_DESKTOP_ALUNO           181 non-null    int64  
 11  QT_DESKTOP_ALUNO           181 non-null    int64  
 12  IN_COMP_PORTATIL_ALUNO     181 non-null    int64  
 13  QT_COMP_PORTATIL_ALUNO     181 non-null    int64  

In [ ]:
df.describe()

CO_UF  CO_MUNICIPIO   CO_ENTIDADE  TP_DEPENDENCIA  TP_LOCALIZACAO  \
count  181.000000  1.810000e+02  1.810000e+02      181.000000      181.000000   
mean    32.779006  3.282505e+06  3.287904e+07        2.756906        1.668508   
std     13.172185  1.318293e+06  1.315163e+07        0.430141        0.472055   
min     11.000000  1.100098e+06  1.102749e+07        2.000000        1.000000   
25%     25.000000  2.501401e+06  2.510310e+07        3.000000        1.000000   
50%     31.000000  3.106507e+06  3.126410e+07        3.000000        2.000000   
75%     42.000000  4.205175e+06  4.208494e+07        3.000000        2.000000   
max     52.000000  5.205307e+06  5.211280e+07        3.000000        2.000000   

       TP_SITUACAO_FUNCIONAMENTO  IN_DESKTOP_ALUNO  QT_DESKTOP_ALUNO  \
count                      181.0        181.000000        181.000000   
mean                         1.0          0.392265          3.314917   
std                          0.0          0.489610          6.690395   
min                          1.0          0.000000          0.000000   
25%                          1.0          0.000000          0.000000   
50%                          1.0          0.000000          0.000000   
75%                          1.0          1.000000          4.000000   
max                          1.0          1.000000         35.000000   

       IN_COMP_PORTATIL_ALUNO  QT_COMP_PORTATIL_ALUNO  ...  i_conec_simet  \
count              181.000000              181.000000  ...           28.0   
mean                 0.237569                2.071823  ...            0.0   
std                  0.426774               14.098476  ...            0.0   
min                  0.000000                0.000000  ...            0.0   
25%                  0.000000                0.000000  ...            0.0   
50%                  0.000000                0.000000  ...            0.0   
75%                  0.000000                0.000000  ...            0.0   
max                  1.000000              177.000000  ...            0.0   

               ano  Competência      versao      i_comp      soma_i  \
count   105.000000   105.000000  105.000000  105.000000  105.000000   
mean   2018.980952     1.933333    1.095238    0.238095    1.133333   
std       0.650162     0.879977    0.427960    0.427960    0.809400   
min    2018.000000     1.000000    1.000000    0.000000    0.000000   
25%    2019.000000     1.000000    1.000000    0.000000    1.000000   
50%    2019.000000     2.000000    1.000000    0.000000    1.000000   
75%    2019.000000     2.000000    1.000000    0.000000    2.000000   
max    2021.000000     4.000000    3.000000    1.000000    3.000000   

           i_viab  apoio_IEC  apoio_CIEB  assessoria_cieb0  
count  105.000000        0.0         0.0               0.0  
mean     0.057143        NaN         NaN               NaN  
std      0.233229        NaN         NaN               NaN  
min      0.000000        NaN         NaN               NaN  
25%      0.000000        NaN         NaN               NaN  
50%      0.000000        NaN         NaN               NaN  
75%      0.000000        NaN         NaN               NaN  
max      1.000000        NaN         NaN               NaN  

[8 rows x 36 columns]

In [ ]:
df[df.index.duplicated()]

Empty DataFrame
Columns: [NO_UF, SG_UF, CO_UF, NO_MUNICIPIO, CO_MUNICIPIO, CO_ENTIDADE, TP_DEPENDENCIA, REDE, TP_LOCALIZACAO, TP_SITUACAO_FUNCIONAMENTO, IN_DESKTOP_ALUNO, QT_DESKTOP_ALUNO, IN_COMP_PORTATIL_ALUNO, QT_COMP_PORTATIL_ALUNO, IN_TABLET_ALUNO, QT_TABLET_ALUNO, IN_INTERNET, IN_INTERNET_ALUNOS, IN_INTERNET_APRENDIZAGEM, TP_REDE_LOCAL, IN_BANDA_LARGA, IN_REGULAR, QT_DOC_BAS, QT_TUR_BAS, QT_MAT_FUND_AI, mat_maior_turno, maior_turno, proxy_maior_turno, TCP_DOWN_MBPS, LAST_MEASURE2, n_disp_est, est_por_disp, nivel_dispos, i_dispos, vel_est, i_conec_simet, ano, Competência, versao, base, data_quest, i_comp, soma_i, i_viab, tam_esc, in_INF, in_FUND_AI, in_FUND_AF, atend_etapas0, apoio_IEC, apoio_CIEB, assessoria_cieb0, Destinação, Vl Devido Custeio, Vl Devido Capital, Vl Devido Total, Vl Ajuste Custeio, Vl Ajuste Capital, Vl Ajuste Total, Vl Final Devido Total, Vl Pago Custeio, Vl Pago Capital, Valor Pago Total, Data Ord. Pgto, in_MED]
Index: []

[0 rows x 65 columns]

In [ ]:
df.head(3)

NO_UF SG_UF  CO_UF NO_MUNICIPIO  CO_MUNICIPIO  CO_ENTIDADE  TP_DEPENDENCIA  \
0  Pará    PA     15   Pau D'Arco       1505551     15130959               3   
1  Pará    PA     15   Pau D'Arco       1505551     15131572               2   
2  Pará    PA     15   Pau D'Arco       1505551     15160467               3   

        REDE  TP_LOCALIZACAO  TP_SITUACAO_FUNCIONAMENTO  ...  Vl Devido Total  \
0  Municipal               2                          1  ...         2.451,00   
1   Estadual               1                          1  ...         3.328,00   
2  Municipal               2                          1  ...                -   

   Vl Ajuste Custeio  Vl Ajuste Capital  Vl Ajuste Total  \
0                  0                  0                0   
1                  0                  0                0   
2                  -                  -                -   

   Vl Final Devido Total  Vl Pago Custeio  Vl Pago Capital  Valor Pago Total  \
0               2.451,00         1.560,01           890,99          2.451,00   
1               3.328,00                0                0                 0   
2                      -                -                -                 -   

   Data Ord. Pgto  in_MED  
0      13/10/2021     NaN  
1             NaN      EM  
2               -     NaN  

[3 rows x 65 columns]

In [ ]:
print("Single column value using dataframe.dot")
print(df.nivel_dispos)


Single column value using dataframe.dot
0      1
1      2
2      1
3      1
4      1
      ..
176    3
177    1
178    1
179    1
180    1
Name: nivel_dispos, Length: 181, dtype: int64


In [ ]:
len(df[df['nivel_dispos'] == 1])

144

144 escolas no nível emergente de dispositivos

In [ ]:
len(df[df['nivel_dispos'] == 2])

13

13 escolas no nível básico

In [ ]:
len(df[df['nivel_dispos'] == 3])

17

17 escolas no nível intermediário

In [ ]:
len(df[df['nivel_dispos'] == 4])

7

7 escolas no nível avançado

24 escolas com pelo menos 5 estudantes por dispositivos

13,2% do total

In [ ]:
df.isnull().sum()

NO_UF                 0
SG_UF                 0
CO_UF                 0
NO_MUNICIPIO          0
CO_MUNICIPIO          0
                   ... 
Vl Pago Custeio       2
Vl Pago Capital       3
Valor Pago Total      3
Data Ord. Pgto       13
in_MED              151
Length: 65, dtype: int64

In [ ]:
len(df[df['Data Ord. Pgto'] == '-'])

100

In [ ]:
df['Data Ord. Pgto'].unique()

array(['13/10/2021', nan, '-', '29/09/2021', '26/11/2021', '09/12/2021',
       '12/11/2021', '16/12/2021', '11/11/2021', '09/11/2021',
       '24/11/2021', '0', '30/09/2021'], dtype=object)

In [ ]:
df["Data Ord. Pgto"].replace("-", np.nan , inplace=True)

In [ ]:
df["Data Ord. Pgto"].replace(0, np.nan , inplace=True)

In [ ]:
df.isnull().sum()

NO_UF                 0
SG_UF                 0
CO_UF                 0
NO_MUNICIPIO          0
CO_MUNICIPIO          0
                   ... 
Vl Pago Custeio       2
Vl Pago Capital       3
Valor Pago Total      3
Data Ord. Pgto      113
in_MED              151
Length: 65, dtype: int64

113 escolas não receberam repasse do PDDE Educação Conectada em 2021

Repasse total em 2021 para as escolas foi de R$ 194.475,00, sendo 103.286,34 para capital e  91.188,66 para custeio

In [ ]:
df['TCP_DOWN_MBPS'].unique()

array([nan, '53,38', '18,77', '82,38', '65,44', '38,63', '36,38', '20,68',
       '22,88', '27,79', '47,24', '42,98', '33,56', '38,07', '13,03',
       '50,85', '9,6', '73,95', '86,61', '63,42', '11,89', '17,88',
       '10,09', '34,26', '9,64', '39,13', '4,74', '11,14', '17,13'],
      dtype=object)

In [ ]:
df['LAST_MEASURE2'].unique()

array([nan, '2/1/22 16:22', '3/4/22 21:05', '3/7/22 6:46', '3/7/22 18:03',
       '3/7/22 16:13', '3/7/22 17:07', '9/15/21 12:08', '12/15/21 11:33',
       '3/7/22 11:05', '3/7/22 15:12', '3/7/22 18:29', '3/7/22 16:23',
       '2/7/22 11:21', '3/3/22 10:22', '3/11/21 6:09', '3/7/22 7:15',
       '10/5/21 8:26', '3/7/22 15:44', '12/24/21 15:12', '12/17/21 1:33',
       '3/7/22 18:41', '8/26/21 10:07', '8/26/21 10:35', '9/30/21 15:27',
       '3/7/22 23:05', '3/5/22 10:26', '3/4/22 8:48', '3/3/22 11:32'],
      dtype=object)

In [ ]:
df.LAST_MEASURE2.isnull().sum()

153

153 escolas sem medidor

In [ ]:
len(df[df['vel_est'] == "NA"])

0

In [ ]:
df['vel_est'].unique()

array([nan, '0,360675676', '0,145503876', '0,206466165', '0,208407643',
       '0,07741483', '0,104240688', '0,144615385', '0,05635468',
       '0,204338235', '0,277882353', '0,172610442', '0,277355372',
       '0,200368421', '0,075755814', '0,225', '0,038709677',
       '0,293452381', '0,241253482', '0,2265', '0,031289474',
       '0,047301587', '0,024371981', '0,283140496', '0,051550802',
       '0,149351145', '0,107727273', '0,179677419', '0,300526316'],
      dtype=object)

In [ ]:
len(df[df['vel_est'] >= "0,1"])

20

20 escolas com 100k/aluno e nenhuma escola no parâmetro de 1 mbps/estudante